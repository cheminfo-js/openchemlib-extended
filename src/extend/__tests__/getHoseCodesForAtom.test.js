'use strict';

const OCLE = require('../..');

describe('getHosesCodeForAtom test propane', () => {
  it('minSphereSize:0, maxSphereSize:3', () => {
    const molecule = OCLE.Molecule.fromSmiles('CCC');
    let results = molecule.getHoseCodesForAtom(0, {
      minSphereSize: 0,
      maxSphereSize: 3
    });
    expect(results).toHaveLength(4);
    results = results.map(result => escape(result));
    expect(results).toMatchSnapshot();
  });

  it('minSphereSize:1, maxSphereSize:2', () => {
    var molecule = OCLE.Molecule.fromSmiles('CCC');
    let results = molecule.getHoseCodesForAtom(0, {
      minSphereSize: 1,
      maxSphereSize: 2
    });
    expect(results).toHaveLength(2);
  });
});
