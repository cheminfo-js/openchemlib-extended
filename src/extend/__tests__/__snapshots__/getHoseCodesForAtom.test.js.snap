// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getHosesCodeForAtom test propane minSphereSize:0, maxSphereSize:3 1`] = `
Array [
  "fH@NJ%60%7FRapj%60",
  "eF@Hp%5CQP_iHNET",
  "eM@Df%60xb%60%7FRP%5CJh",
  "eM@Df%60xb%60%7FRP%5CJh",
]
`;
